### PR TITLE
Sampler callback checks parent decision

### LINKED ISSUE #3230: Examples of `TracesSampler` should check for parent sampling decision
We document that:

> A boolean `parentSampled` which contains the sampling decision passed down from the parent, if any..

This is a bullet point in the docs and easily missed. The side effect of missing this information results in broken traces, significantly affecting customer.

An immediate improvement we can make is to change all our code samples for `TracesSampler` to show case a check for the SamplingContext and taking this information into account when making a decision. A code comment above explaining why this is important should also be included.

Affected snippets are:

https://github.com/getsentry/sentry-docs/tree/master/src/includes/performance/traces-sampler-as-sampler

See also: https://github.com/getsentry/sentry-java/issues/1315

### PR DESCRIPTION
**Pull request by bruno-garcia**
Resolves #3230

SDKs (https://github.com/getsentry/sentry-docs/tree/master/src/includes/performance/traces-sampler-as-sampler):

- [x] Java
- [ ] Spring
- [ ] Spring Boot
- [x] .NET
- [ ] Ruby
- [ ] PHP
- [ ] Python
- [ ] Node
- [ ] Go

halp @maciejwalkowiak @Tyrrrz @st0012 @kamilogorek @ahmedetefy @stayallive @lobsterkatie

### CODE REVIEW

**Comment by st0012** [comment]
@bruno-garcia currently the Ruby SDK doesn't perform sampling correctly and will be fixed shortly. so I'll update the document after the fix is released.

File: src/includes/performance/traces-sampler-as-sampler/java.mdx [hunk 1]
```diff
@@ -10,6 +10,10 @@ Sentry.init(options -> {
   options.setTracesSampler(context -> {
     CustomSamplingContext ctx = context.getCustomSamplingContext();
     if (ctx != null) {
+      // If this is the continuation of a trace, just use that decision (rate controlled by the caller).
+      if (ctx.getParentSampled() != null) {
+        return ctx.getParentSampled();
+      }
```
Comment (marandaneto) [inline src/includes/performance/traces-sampler-as-sampler/java.mdx:16]: I believe you meant another object here:

```
val transactionContext = it.transactionContext
transactionContext.parentSampled
```

Comment (bruno-garcia) [inline src/includes/performance/traces-sampler-as-sampler/java.mdx:16]:
```suggestion
      // If this is the continuation of a trace, just use that decision (rate controlled by the caller).
      Boolean parentSampled = context.getTransactionContext().getParentSampled();
      if (parentSampled != null) {
        return parentSampled ? 1.0 : 0.0;
```

File: src/includes/performance/traces-sampler-as-sampler/java.mdx [hunk 2]
```diff
@@ -10,6 +10,10 @@ Sentry.init(options -> {
   options.setTracesSampler(context -> {
     CustomSamplingContext ctx = context.getCustomSamplingContext();
     if (ctx != null) {
+      // If this is the continuation of a trace, just use that decision (rate controlled by the caller).
+      Boolean parentSampled = context.getTransactionContext().getParentSampled();
+      if (parentSampled != null) {
+        return parentSampled ? 1.0 : 0.0;
```
Comment (bruno-garcia) [inline src/includes/performance/traces-sampler-as-sampler/java.mdx:16]:
```suggestion
        return parentSampled ? 1.0 : 0.0;
      }
```

**Comment by Tyrrrz** [comment]
@Tyrrrz tagging myself

**Comment by marandaneto** [comment]
@bruno-garcia I believe we should split up PRs and leave this one as a place holder otherwise we won't merge it, like not so soon

**Comment by marandaneto** [comment]
@maciejwalkowiak would u like to go ahead and add the parent sampling examples to Java and Android docs following @bruno-garcia example? thanks

**Comment by maciejwalkowiak** [comment]
Ok

**Comment by maciejwalkowiak** [comment]
I did more changes here: https://github.com/maciejwalkowiak/sentry-docs/tree/feat/trace-sampling-check-parent-java
@bruno-garcia either please cherry pick relevant commits, or please pull changes from master to `feat/trace-sampling-check-parent`

**Comment by bruno-garcia** [comment]
I'll merge this. @maciejwalkowiak feel free to open the spring PR straight to master

**Comment by bruno-garcia** [comment]
Thanks @marandaneto @maciejwalkowiak @Tyrrrz

**Pull request merged**